### PR TITLE
Fix assert issue

### PR DESCRIPTION
**Pull request by kevinwhereby**
see original PR that I accidentally merged here: https://github.com/whereby/browser-sdk/pull/79
and the revert here: https://github.com/whereby/browser-sdk/pull/80
🤦 

### Description

We are having some funny module resolution problems with rollup, so this pr aims to replace it with esbuild

Adds build steps for individual react/embed builds as well

### Gotchas
Esbuild can't emit typescript declarations, so we have to do it with `tsc`
Unfortunately, [tsc can't generate es6 concatenated declarations](https://github.com/microsoft/TypeScript/issues/2568) so we include the full directory structure... is this a problem? 

### To Test
- `yarn` to install the stuff
- `yarn build` to generate the builds and types
-  import it into a demo project

### Screenshots
Here's me importing just the react bundle like `import { useRoomConnection } from "@whereby.com/browser-sdk/react"`
image

and using the embed bundle in a script tag like
`<script type="module" src="./node_modules/@whereby.com/browser-sdk/dist/embed/index.esm.js"></script>`
image

### CODE REVIEW

**Comment by kevinwhereby** [comment]
@havardholvik oops sorry

**Comment by kevinwhereby** [comment]
@havardholvik re. the /react and /embed imports, what do you think about moving that to a different PR considering everything else seems to be working here?

**Comment by havardholvik** [comment]
> @havardholvik re. the /react and /embed imports, what do you think about moving that to a different PR considering everything else seems to be working here?

Im fine with that, as long as we see an improvement (and no regressions) with what we have today 👍

**Comment by kevinwhereby** [comment]
Ah I lied, CRA still doesn't work..

**Comment by kevinwhereby** [comment]
OK this _finally_ works on CRA and Vite as an import and a script tag referencing the bundled `embed.esm.js` file (unless I'm mistaken)

**Review by havardholvik — CHANGES_REQUESTED** [review]
If this PR aims at solving the assert-related issue we are seeing; I'm still getting the `process is not defined` error when trying out this build in Vite
Screenshot 2023-10-16 at 08 50 37

**Comment by kevinwhereby** [comment]
@havardholvik wanna compare configs?
vite 4.4.5
react 18.2
typescript 5.02
```javascript
// vite.config.js
export default defineConfig({
    plugins: [react()],
    define: {
        "process.env": {},
    },
    build: {
        minify: false,
    },
    server: {
        port: 6006,
    },
});
```

```javascript
//tsconfig
{
    "compilerOptions": {
        "target": "es2016",
        "useDefineForClassFields": true,
        "lib": ["esnext", "DOM", "DOM.Iterable"],
        "module": "esnext",
        "skipLibCheck": true,

        /* Bundler mode */
        "moduleResolution": "bundler",
        "allowImportingTsExtensions": true,
        "resolveJsonModule": true,
        "isolatedModules": true,
        "noEmit": true,
        "jsx": "react-jsx",

        /* Linting */
        "strict": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noFallthroughCasesInSwitch": true
    },
    "include": ["src"],
    "references": [{ "path": "./tsconfig.node.json" }]
}
```

**Review by havardholvik — APPROVED** [review]
Vite works for me now once the process.env is defined 👍  Make sure also that we support our own environment variables, I think @nandito have been using them to eg override signal and api host.

File: build.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    .concat(Object.keys(pkg.peerDependencies))
+    .filter((dep) => !dep.match(/jslib-media|events/));
+
+const sharedConfig = {
+    bundle: true,
+    external,
+    format: "esm",
+    minify: process.env.NODE_ENV === "production",
+    platform: "browser",
+    target: ["es2016"],
```
Comment (havardholvik) [inline build.js:21]: I know this was my recommendation, but seeing we target es6 in our tsconfig, I think we should use that here as well

**Pull request merged**